src/main/java/bio/terra/app/logging/PerformanceLoggerON.java

```diff
@@ -0,0 +1,128 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Primary
+@Profile({"perftest"})
+@Component
+
+/**
+ * This class implements ENABLED performance logging. It is used when the "perftest" profile is active.
+ *
+ * It generates logs using the same slf4j logger used throughout the API code, but with a specific
+ * format intended to be propagated from StackDriver to BigQuery and then parsed with SQL for further analysis.
+ *
+ * Any time there is a new type of information that we want to include in performance logs, there are 2 options:
+ *   - Add the new field to the format string in this class and add optional arguments to the log methods that accept
+ *     a value for this new field.
+ *   - Specify the new information as a serializable object (e.g. String or property of a POJO) and pass it to the
+ *     existing log methods with the additionalInfo parameter.
+ *  New types of information that seem broadly useful should use the first option and those that will probably only
+ *  be used in a small number of places should use the second option.
+ */
+public class PerformanceLoggerON implements PerformanceLogger {
+
+    public static final boolean isPerformanceLoggingEnabled = true;
+
+    private ObjectMapper objectMapper;
+
+    private static Logger logger = LoggerFactory.getLogger(PerformanceLoggerON.class);
+    private static final String PerformanceLogFormat =
+        "TimestampUTC: {}, JobId: {}, Class: {}, Operation: {}, " +
+        "ElapsedTime: {}, IntegerCount: {}, AdditionalInfo: {}";
+
+    // Implementation choice note:
+    // Using a thread local map for timers instead of a static concurrent hash map avoids locks when updating the map.
+    // Thread local means we can't track timers across different threads, while a concurrent hash map would allow this.
+    // Neither option allows tracking timers across different DR Managers. We could use SQL to calculate time durations
+    // between start/end timestamps to address that use case.
+    private static final ThreadLocal<Map<String, Instant>> startTimeMap =
+        ThreadLocal.withInitial(() -> new HashMap<>());
+
+    @Autowired
+    public PerformanceLoggerON(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+        logger.info("Performance logging ON");
+    }
+
+    public void log(String jobId, String className, String operationName,
+                    long elapsedTime, long integerCount, Serializable additionalInfo) {
```

Comment:
Serializable gives no guarantee that the object is JSON-able.
Is there a reason not to just log the `toString()`?

Reply:
You're right, I was mixing Serializable and JSON-able. Updated to take an Object and log the string anyway, even if it doesn't serialize into JSON properly.